wlp-managed-8.5/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -856,6 +860,37 @@ public void undeploy(Descriptor descriptor) throws DeploymentException {
 
    }
 
+   private void checkForDefinitionExceptions(String applicationName)
+   {
+      String messagesFile = containerConfiguration.getWlpHome() + "/usr/servers/" + containerConfiguration.getServerName() + "/logs/messages.log";
+      BufferedReader br = null;
+
+      try {
+         br = new BufferedReader(new InputStreamReader(new FileInputStream(messagesFile)));
+         String line;
+         while ((line = br.readLine()) != null) {
+            if (line.contains("CWWKZ0002E: An exception occurred while starting the application " + applicationName + ".")
+                  && (line.contains("org.jboss.weld.exceptions.DefinitionException") || line.contains("javax.enterprise.inject.spi.DefinitionException"))) {
+               System.out.println("############DEBUG found CWWKZ0002E for application: " + applicationName);
```

Comment:
We will have to change these to be written to the logger only in log.finest()

Reply:
OK.